lab_4_retrieval_w_clustering/main.py

```diff
@@ -170,6 +239,8 @@ def __init__(self, db: DocumentVectorDB) -> None:
         Args:
             db (DocumentVectorDB): Object of DocumentVectorDB class.
         """
+        self._db = db
+        super().__init__(self._db.get_vectorizer(), self._db.get_tokenizer())
```

Comment:
better to call super as soon as possible.

Reply:
fixed